### PR TITLE
Feature/draft preview

### LINKED ISSUE #9: Add draft preview
**Is your feature request related to a problem? Please describe.**
As a user, I want to be able to preview a draft on my website.
I want to open the preview from the Hashnode editor and have it open on my website.

**Describe the solution you'd like**
Get the draft from Hashnode and render it like a regular blog post.

**Additional context**
This is already implemented in the Hashnode/starter-kit

### PR DESCRIPTION
**Pull request by lkrimphove**
## What does this PR do?

Fixes #9

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Chore (refactoring code, technical debt, workflow improvements)
- [ ] Enhancement (small improvements)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## How should this be tested?

- Create a draft in the Hashnode editor.
- Open the preview of this draft.
- The draft should be shown on your website.

### Things to remember

- [X] Filled out the "How to test" section in this PR
- [X] Read [Contributing Guide](./CONTRIBUTING.md)
- [X] Ran `pnpm build`
- [X] Ran `pnpm format`
- [X] Ran `pnpm lint`
- [X] Checked for warnings, there are none
- [X] Removed all `console.logs`
- [X] Merged the latest changes from main onto my branch with `git pull origin main`

### CODE REVIEW

**Comment by alexkates** [comment]
@lkrimphove can you address the merge conflicts in `generated/graphql.ts`?

I've created a new issue to [exclude the generated files](https://github.com/alexkates/hashnode-next/issues/11) from git.

File: src/app/blog/preview/[id]/page.tsx [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+import { Metadata } from "next/types";
+
+type Props = {
+  params: {
+    id: string;
+  };
+};
+
+export async function generateMetadata({ params }: Props) {
+  const post = await getdraftDraft(params);
```
Comment (alexkates) [inline src/app/blog/preview/[id]/page.tsx]: getBlogPostDraft*

**Review by alexkates — CHANGES_REQUESTED** [review]
One minor name change and then it's good to go!

File: src/app/blog/preview/[id]/page.tsx (hunk 1 above)
Comment (lkrimphove) [reply src/app/blog/preview/[id]/page.tsx]: Sorry for those mistakes. Fixed those and tested it again. Everything should work now. :)

**Comment by alexkates** [comment]
Awesome! Thanks again @lkrimphove

**Pull request merged**